flake.nix

```diff
@@ -3,32 +3,42 @@
   
   inputs = {
     nixpkgs.url = "github:NixOS/nixpkgs/nixos-24.05";
+    circom-compat.url = "github:codex-storage/circom-compat-ffi";
   };
 
-  outputs = { self, nixpkgs }:
+  outputs = { self, nixpkgs, circom-compat}:
     let
-      supportedSystems = [
+      stableSystems = [
         "x86_64-linux" "aarch64-linux"
         "x86_64-darwin" "aarch64-darwin"
       ];
-      forAllSystems = f: nixpkgs.lib.genAttrs supportedSystems (system: f system);
-      pkgsFor = forAllSystems (system: import nixpkgs { inherit system; });
+      forEach = nixpkgs.lib.genAttrs;
+      forAllSystems = forEach stableSystems;
+      pkgsFor = forEach stableSystems (
+        system: import nixpkgs { inherit system; }
+      );
     in rec {
+      packages = forAllSystems (system: let
+        circomCompatPkg = circom-compat.packages.${system}.default;
+        buildTarget = pkgsFor.${system}.callPackage ./nix/default.nix {
+          inherit stableSystems circomCompatPkg;
+          src = self;
+        };
+        build = targets: buildTarget.override { inherit targets; };
+      in rec {
+        codex   = build ["all"];
+        default = codex;
+      });
+
       devShells = forAllSystems (system: let
         pkgs = pkgsFor.${system};
-        inherit (pkgs) lib stdenv mkShell;
       in {
-        default = mkShell.override { stdenv = pkgs.gcc11Stdenv; } {
-          buildInputs = with pkgs; [
-              # General
-              git pkg-config openssl lsb-release
-              # Build
-              rustc cargo nimble gcc11 cmake nim-unwrapped-1
-              # Libraries
-              gmp llvmPackages.openmp
-              # Tests
-              nodejs_18
+        default = pkgs.mkShell {
+          inputsFrom = [
+            packages.${system}.codex
+            circom-compat.packages.${system}.default
           ];
+          buildInputs = with pkgs; [ git nodejs_18 ];
```

Comment:
There is one issue with this approach, and that `buildInputs` dome in after `inputsFrom` in `PATH`:
```
[jakubgs@caspair:~/work/nim-codex]$ echo $PATH | tr ':' '\n' | grep git
/nix/store/cn0hd2vyrin50vsqqam6xy5jlqdgal8a-git/bin
/nix/store/vdd8ccp9x69v7kp5g9khq7j43n6dqf16-git-2.44.1/bin
```
So `fakeGit` actually wins due to the order of search. Which is annoying, but can be fixed by using `nativeBuildInputs`:
```suggestion
          # Not using buildInputs to override fakeGit and fakeCargo.
          nativeBuildInputs = with pkgs; [ git cargo nodejs_18 ];
```